scripts/subsampling_PLIER.R

```diff
@@ -0,0 +1,208 @@
+#    seed: initial seed (default is 123)
+#    use_sample_list: whether or not a file with a list of sample names should
+#                     be used for the subsetting (default is FALSE); if TRUE
+#                     no repeats will be performed
+#    sample_list: TSV file that contains a single column, without a header, of
+#                 the samples to be included in the training set
+
+#### custom functions ----------------------------------------------------------
+
+PLIERWrapper <- function(exprs, pathway.mat, seed = 12345) {
+  # "Pared down" PLIER wrapper function. The one we use elsewhere includes
+  # collapsing to common genes between the expression matrix and the gene sets
+  # that are included with the PLIER package -- we already have this information
+  # for this experiment
+  #
+  # Args:
+  #   exprs: a gene expression matrix, rows are genes, columns are samples
+  #   pathway.mat: combined pathway matrix for use with PLIER, same
+  #                genes and ordering as exprs
+  #   seed: an integer to be supplied to set.seed() for reproducibility 
+  #         purposes, default is 12345
+  # 
+  # Returns
+  #   a list that contains the row-normalized expression data and the PLIER
+  #   output
+  
+  # trouble with glmnet otherwise
+  library(PLIER)
+  
+  set.seed(seed)
+  
+  # check row order of expression and pathway matrices
+  ord.row.check <- all.equal(rownames(exprs), rownames(pathway.mat))
+  if (ord.row.check != TRUE) {
+    stop("rownames for exprs and pathway.mat should be the same.")
+  }
+  
+  # rescale
+  exprs <- PLIER::rowNorm(exprs)
+  
+  # what should we set the minimum k parameter to in PLIER? estimate the number
+  # of PC for the SVD decomposition
+  set.k <- PLIER::num.pc(exprs)
+  
+  # PLIER main function + return results
+  plier.res <- PLIER::PLIER(as.matrix(exprs), as.matrix(pathway.mat),
+                            k = round((set.k + set.k * 0.3), 0), trace = TRUE)
+  return(list("exprs" = exprs,
+              "PLIER" = plier.res))
+  
+}
+
+#### parse command line arguments ----------------------------------------------
+
+# loading the library seems a bit cleaner here
+library("optparse")
+
+option.list = list(
+  make_option(c("-i", "--input"), type = "character", default = NULL,
+              help = "Prepped expression and pathway matrices (.RDS)"),
+  make_option(c("-o", "--output"), type = "character", default = NULL,
+              help = "Output file (with full path) for models (.RDS)"),
+  make_option(c("-n", "--num_samples"), type = "integer", default = 500,
+              help = "Sample size for subsampling"),
+  make_option(c("-r", "--repeats"), type = "integer", default = 5,
+              help = "Number of repeats to perform"),
+  make_option(c("-s", "--seed"), type = "integer", default = 123,
+              help = "Number of repeats to perform"),
+  make_option(c("-u", "--use_sample_list"), type = "logical", default = FALSE,
```

Comment:
for a logical option, can use argument `action='store_true'`.

This way, when its called [here](https://github.com/greenelab/multi-plier/pull/45/files#diff-c1c279edbb018e9fd0183d3e08065eadR27) all that is needed is `--use_sample_list` instead of `--use_sample_list TRUE`